src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,631 @@
+///  \brief         Contains SVD components: Eigenvalues and Eigenvectors
+template <typename T>
+class SVDComponentsAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+
+    T const                                          SingularValues;
+    T const                                          SingularVectors;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SVDComponentsAnnotationData(T singularvalues, T singularvectors);
+    ~SVDComponentsAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(SVDComponentsAnnotationData);
+};
+
+namespace ComponentsDetails {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SVDTrainingOnlyPolicy
+///  \brief         `SVDComponentsEstimator` implementation details.
+///
+template <typename InputT, typename TransformedT>
+class SVDTrainingOnlyPolicy {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                                  = InputT;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const    NameValue = SVDComponentsEstimatorName;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    void fit(InputType const &input);
+    SVDComponentsAnnotationData<TransformedT> complete_training(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    TransformedT                                     _matrix;
+    TransformedT                                     _singularvalues;
+    TransformedT                                     _singularvectors;
+};
+
+} // namespace ComponentsDetails
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SVDComponentsEstimator
+///  \brief         This class generates SVDComponentsAnnotationData
+///                 by processing input matrix using Eigen
+///
+template <
+    typename InputT,
+    typename TransformedT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class SVDComponentsEstimator :
```

Comment:
Could the functionality in this Estimator be implemented in the Impl class? If so, you can remove this estimator and eliminate the use of the PipelineExecutionEstimatorImpl.